mount/infoParser.go

```diff
@@ -472,3 +450,109 @@ func (mi *mountInfoParser) HasNonSysboxfsSubmount(basemount string) bool {
 
 	return false
 }
+
+// IsRoMount checks if the passed mountpoint is currently present and tagged as
+// read-only.
+func (mi *mountInfoParser) IsRoMount(mountpoint string) bool {
+
+	info := mi.GetInfo(mountpoint)
+	if info == nil {
+		return false
+	}
+
+	perMountFlags := mi.service.mh.StringToFlags(info.Options)
+
+	return perMountFlags&unix.MS_RDONLY == unix.MS_RDONLY
+}
+
+// IsRecursiveBindMount verifies if the passed mountinfo entry is a recursive
+// bind-mount.
+//
+// Example: [ mouuntID-3413 is a recursive bind-mount of mountID-3544 ]
+//
+// 3544 3503 0:129 / /usr/src/linux-headers-5.4.0-48 ro,relatime - shiftfs /usr/src/linux-headers-5.4.0-48 rw
+// 3413 3544 0:129 / /usr/src/linux-headers-5.4.0-48 ro,relatime - shiftfs /usr/src/linux-headers-5.4.0-48 rw
+func (mi *mountInfoParser) IsRecursiveBindMount(info *domain.MountInfo) bool {
+
+	devIdSlice := mi.devIdInfo[info.MajorMinorVer]
+
+	for _, elem := range devIdSlice {
+		if elem.MountID == info.MountID {
+			continue
+		}
+
+		if elem.MountID == info.ParentID &&
+			elem.Source == info.Source &&
+			elem.Root == info.Root {
+			return true
+		}
+	}
+
+	return false
+}
+
+// IsBindMount verifies if the passed mountinfo entry is a bind-mount.
+func (mi *mountInfoParser) IsBindMount(info *domain.MountInfo) bool {
+
```

Comment:
Suggestion: let's add a comment here with a high level description on how we discern that the given mount is a bind-mount. That way we won't have to re-think this in the future.